typescript/sdk/src/consts/chainMetadata.ts

```diff
@@ -1,187 +1,606 @@
-import { ChainName } from '../types';
+import type { Chain as WagmiChain } from '@wagmi/chains';
+
+import { objMap } from '../utils/objects';
+
+import { ChainName, Chains, Mainnets, Testnets } from './chains';
+
+export enum ExplorerFamily {
+  Etherscan = 'etherscan',
+  Blockscout = 'blockscout',
+  Other = 'other',
+}
 
 /**
- * A Chain and its characteristics
+ * Collection of useful properties and settings
+ * for Hyperlane-supported chains
  */
-export type ChainMetadata = {
+export interface ChainMetadata {
   id: number;
-  finalityBlocks: number;
-  nativeTokenDecimals?: number;
-  paginate?: RpcPagination;
-  // The CoinGecko API expects, in some cases, IDs that do not match
-  // ChainNames.
+  name: ChainName;
+  /** Human-readable name */
+  displayName: string;
+  /** Shorter human-readable name */
+  displayNameShort?: string;
+  /** Default currency/token used by chain */
+  nativeToken: {
+    name: string;
+    symbol: string;
+    decimals: number;
+  };
+  /** Collection of RPC endpoints */
+  publicRpcUrls: Array<{
+    http: string;
+    webSocket?: string;
+    pagination?: RpcPagination;
+  }>;
+  /** Collection of block explorers */
+  blockExplorers: Array<{
+    name: string;
+    url: string;
+    family: ExplorerFamily;
+    apiUrl?: string;
+  }>;
+  blocks: {
+    // Number of blocks to be considered final
+    finalityConfirmations: number;
```

Comment:
I wonder if this should just be "confirmations: Number of blocks to wait before considering a transaction confirmed"

If you add reorgPeriod (or alternatively "finality"), consider "The number of blocks before a transaction has a near zero chance of reverting"

Reply:
Yep that sgtm